### PR TITLE
Relogin popup

### PR DESCRIPTION
**Pull request by OkapalDominic**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Adds a popup to login when session has expired.

* **What is the current behavior?** (You can also link to an open issue here)
Can't do anything if your session expires.

* **What is the new behavior (if this is a feature change)?**
If you attempt a server request and it returns a 401 error, a popup appears, allowing you to re-login.

* **Other information**:
Created a service that does what server-api service does to avoid circular imports.  Moved baseUrl to a common file so that it's not necessary to put it in both places.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -88,13 +88,16 @@
     "@progress/kendo-angular-pdf-export": "^1.0.1",
     "@progress/kendo-drawing": "^1.4.2",
     "angular2-text-mask": "^8.0.4",
+    "body-parser": "^1.19.0",
     "bootstrap": "^3.3.7",
     "bootstrap-loader": "^2.2.0",
     "bootstrap-sass": "^3.3.7",
     "classlist.js": "^1.1.20150312",
     "core-js": "^2.4.1",
+    "cors": "^2.8.5",
```
Comment (ElijahRW) [inline package.json:97]: I don't think this should not be added into package.json

File: src/app/relogin-popup/relogin-popup.service.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+                    Utilities.setSessionStorage('sessionId', response.data.session.toString());
+                    console.log(authorityTypes[response.data.authorityType]);
+                    Utilities.setSessionStorage('permissions', authorityTypes[response.data.authorityType]);
+                }),
+                map<Response<LoginSession>, null>(_ => null)
+            );
+    }
+
+    private post<T>(urlExtension: string, body: any, queryParams?: Map<string, string>): Observable<Response<T>> {
+        return this.http.post<Response<T>>(baseUrl + urlExtension, body, this.createHeaders(queryParams));
```
Comment (wynnliam) [inline src/app/relogin-popup/relogin-popup.service.ts:34]: Everything seems okay to me. The only thing I'll suggest is a comment or something better explain what this is for. Even in the context of ReloginPopupService, it's not clear to me what the need is for this post. Still though, not something that would prevent a merge in my opinion.

**Review by wynnliam — APPROVED** [review]

File: src/app/relogin-popup/relogin-popup.service.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+import { LoginSession } from '../../../api-objects/LoginSession';
+import { authorityTypes } from '../../../api-objects/UsersObject';
+import baseUrl from '../common/base-url';
+
+@Injectable()
+export class ReloginPopupService {
+
+    constructor(private http: HttpClient) { }
+
+    public login(username: string, password: string): Observable<null> {
```
Comment (ElijahRW) [inline src/app/relogin-popup/relogin-popup.service.ts:19]: I think you should call use the login service already implemented to set up session storage.

**Review by ElijahRW — APPROVED** [review]
Looks good, but maybe use the login service already written for the onLogin() function.

**Pull request merged**